backend/src/main/java/com/google/minicrm/data/Lead.java

```diff
@@ -209,6 +245,74 @@ public static Key generateKey(Key parentKey, String leadId) {
     return KeyFactory.createKey(parentKey, KIND_NAME, leadId);
   }
 
+  /**
+   * Searches for any location data present and populates the estimatedLongitude and
+   * estimateLatitude instance variables.
+   */
+  private void generateLocationInfo() {
+    StringBuilder locationInfo = new StringBuilder();
+    String phoneNumber = "";
+
+    for (String key : columnData.keySet()) {
+      // If the column data is related with address add it into locationInfo arraylist
+      if (key.equals("POSTAL_CODE") || key.equals("STREET_ADDRESS") ||
+          key.equals("CITY") || key.equals("REGION") ||
+          key.equals("COUNTRY") || key.equals("COUNTRY")) {
+        locationInfo.append(columnData.get(key) + " ");
+      } else if (key.equals("PHONE_NUMBER") ) {
+        phoneNumber = columnData.get("PHONE_NUMBER");
+      }
+    }   
+    if (locationInfo.length() == 0) {
+     final int areaCodeFromLead = Integer.parseInt(phoneNumber.substring(0, 3));
```

Comment:
Does this work for phone numbers with international codes: i.e. +1 123-456-7890 instead of 123-456-7890

Reply:
Also, how do we know that there is a phoneNumber here? what if there is no locationInfo and no phoneNumber?

Reply:
Also partially related, can we assume phone number input will be sanitized? (i.e. only numbers and no dashes, spaces, plus signs, etc)

Reply:
We are assuming phone numbers are sanitized for now because that is probably handled by the Form Ads sides

Reply:
If we have time we can add more areaCodes